CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/GalleryFragment.kt

```diff
@@ -102,54 +113,59 @@ class GalleryFragment internal constructor() : Fragment() {
 
         // Handle share button press
         view.findViewById<ImageButton>(R.id.share_button).setOnClickListener {
-            // Make sure that we have a file to share
-            mediaList.getOrNull(mediaViewPager.currentItem)?.let { mediaFile ->
-
-                // Create a sharing intent
-                val intent = Intent().apply {
-                    // Infer media type from file extension
-                    val mediaType = MimeTypeMap.getSingleton()
-                            .getMimeTypeFromExtension(mediaFile.extension)
-                    // Get URI from our FileProvider implementation
-                    val uri = FileProvider.getUriForFile(
-                            view.context, BuildConfig.APPLICATION_ID + ".provider", mediaFile)
-                    // Set the appropriate intent extra, type, action and flags
-                    putExtra(Intent.EXTRA_STREAM, uri)
-                    type = mediaType
-                    action = Intent.ACTION_SEND
-                    flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
-                }
-
-                // Launch the intent letting the user choose which app to share with
-                startActivity(Intent.createChooser(intent, getString(R.string.share_hint)))
+
+            val mediaFile = mediaList[(mediaViewPager.currentItem)]
+
+            // Create a sharing intent
+            val intent = Intent().apply {
+                // Infer media type from file extension
+                val mediaType = MimeTypeMap.getSingleton()
+                        .getMimeTypeFromExtension(mediaFile.extension)
+                // Get URI from our FileProvider implementation
+                val uri = FileProvider.getUriForFile(
+                        view.context, BuildConfig.APPLICATION_ID + ".provider", mediaFile)
+                // Set the appropriate intent extra, type, action and flags
+                putExtra(Intent.EXTRA_STREAM, uri)
+                type = mediaType
+                action = Intent.ACTION_SEND
+                flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
             }
+
+            // Launch the intent letting the user choose which app to share with
+            startActivity(Intent.createChooser(intent, getString(R.string.share_hint)))
         }
 
         // Handle delete button press
         view.findViewById<ImageButton>(R.id.delete_button).setOnClickListener {
+
+            val mediaFile = mediaList[(mediaViewPager.currentItem)]
```

Comment:
There's no need for the parenthesis, you can just say `mediaList[mediaViewPager.currentItem]`